### DISCUSSION TITLE
Adding request ID to any errors that might happen

### DISCUSSION THREAD

**Discussion by sebastian-correa** [fastapi/fastapi#10230] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI
from contextvars import ContextVar
import uuid

app = FastAPI()

uuid_context = ContextVar("uuid", default=None)

@app.middleware("http")
async def add_uuid_middleware(request: Request, call_next: Callable[[Any], Any]) -> Any:
    # Set request UUID.
    uuid_context.set(uuid.uuid4())
    return await call_next(request)


@app.get("/endpoint/")
def endpoint():
    raise ValueError()
```

### Description

I would like to be able to return the value of `uuid_context` in the exception the response that `FastAPI` returns when **any** error occurs in my endpoint function (this could be from a library I control, a 3rd party library or a direct exception risen in the endpoint itself).

I don't have a hard requirement for the response format, I just want to _add_ the ID to whatever FastAPI already returns. I don't want to have to make a response myself, or assign any codes to the return, as FastAPI does a better job than I'd do at this.

### Operating System

macOS

### Operating System Details

MacOS Ventura 13.3.1 (22E261)

### FastAPI Version

0.99.1

### Pydantic Version

1.10.12

### Python Version

Python 3.10.8

**Comment by sebastian-correa** [comment]
Bumping my question as it hasn't been acknowledged :S. Am I missing context to get any help?

**Comment by Kludex** [comment, marked as answer]
https://github.com/snok/asgi-correlation-id

**Reply by sebastian-correa** [reply]
Thanks. Will give it a shot!
